sprox/sa/provider.py

```diff
@@ -95,26 +102,14 @@ def get_fields(self, entity):
         return field_names
 
     def get_entity(self, name):
-        for mapper in _mapper_registry:
-            if mapper.class_.__name__ == name:
-                engine = mapper.tables[0].bind
-                if engine is not None and mapper.tables[0].bind != self.engine:
-                    continue
-                return mapper.class_
-        raise KeyError('could not find model by the name %s'%(name))
+        return mapped_class(self.engine, name)
```

Comment:
What about the previous `raise KeyError('could not find model by the name %s'%(name))`  error?

Reply:
`mapped_class` raises the KeyError instead